src/components/editors.tsx

```diff
@@ -61,10 +61,7 @@ export const DocumentEditor: React.FC<IProps> = ({
             action,
             operation => {
                 window.documentEditorDebugTools?.pushOperation(operation);
-
-                if (operation) {
```

Comment:
I think if an action throws an error, the resulting operation is `undefined`

Reply:
```
 setState(_state => {
            if (!documentReducer) return _state;

            try {
                const newState = documentReducer(_state, action);
                const scope = action.scope ?? 'global';
                const operations = newState.operations[scope];
                const operation = operations[operations.length - 1];

                callback?.(operation);

                return newState;
            } catch (error) {
                onErrorHandler(error);
                onErrorCallback?.(error);
                return _state;
            }
        });
```
This function called on the `callback?.(operation);` line, so it's only called if the operation is found.

Reply:
I see, aweomse!